data/json/portal_storm_effect_on_condition.json

```diff
@@ -136,16 +136,47 @@
       }
     },
     "effect": [
+      { "u_location_variable": { "global_val": "portal_storm_center" }, "min_radius": 30, "max_radius": 75 },
       { "arithmetic": [ { "global_val": "var", "var_name": "cause_early_portal_storm" }, "=", { "const": 1 } ] },
       "next_weather",
-      { "queue_eocs": "EOC_CAUSE_PORTAL_STORM", "time_in_future": [ "2 minutes", "20 minutes" ] },
+      { "queue_eocs": "EOC_CAUSE_PORTAL_STORM", "time_in_future": [ "30 seconds", "1 minutes" ] },
+      { "assign_mission": "MISSION_INVESTIGATE_PORTAL_STORM_CENTER" },
       {
-        "u_message": "You suddenly register a buzzing in your senses.  It's getting louder, and your head starts to throb.  Something is coming.",
+        "u_message": "You suddenly register a buzzing in your senses.  It's getting louder, and your head starts to throb.  Somewhere nearby a mini cataclysm has begun(check mission log for details).",
         "type": "bad",
         "popup": true
       }
     ]
   },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_PORTAL_STORM_CHECK",
+    "//": "Normally weather is only checked every 5 minutes or so, since your distance to the storm center can change quickly we want to test much more often.",
+    "recurrence": [ "1 seconds", "5 seconds" ],
+    "global": true,
+    "condition": { "or": [ { "math": [ "cause_early_portal_storm", "==", "1" ] }, { "math": [ "cause_portal_storm", "==", "1" ] } ] },
+    "deactivate_condition": { "and": [ { "math": [ "cause_early_portal_storm", "!=", "1" ] }, { "math": [ "cause_portal_storm", "!=", "1" ] } ] },
+    "effect": "next_weather"
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_MOVE_PORTAL_STORM",
+    "recurrence": [ "45 seconds", "90 seconds" ],
+    "global": true,
+    "condition": { "or": [ { "math": [ "cause_early_portal_storm", "==", "1" ] }, { "math": [ "cause_portal_storm", "==", "1" ] } ] },
+    "deactivate_condition": { "and": [ { "math": [ "cause_early_portal_storm", "!=", "1" ] }, { "math": [ "cause_portal_storm", "!=", "1" ] } ] },
+    "effect": [
+      {
+        "location_variable_adjust": { "global_val": "portal_storm_center" },
+        "overmap_tile": true,
+        "x_adjust": [ -1, 1 ],
+        "y_adjust": [ -1, 1 ]
+      },
+      { "u_message": "The portal storm shifts!", "type": "bad" },
```

Comment:
I'm a little worried this could get annoying.

Reply:
Fair enough, I've removed the message.